packages/docs/static/data/medplumDefinitions/accesspolicy.json

```diff
@@ -239,48 +239,16 @@
     {
       "name": "writeCriteria",
       "depth": 2,
-      "types": [
-        {
-          "datatype": "BackboneElement"
-        }
-      ],
-      "path": "AccessPolicy.resource.writeCriteria",
-      "min": 0,
-      "max": "1",
-      "short": "",
-      "definition": "Invariants that must be satisfied for the resource to be written.",
-      "comment": "",
-      "inherited": false
-    },
-    {
-      "name": "pre",
-      "depth": 3,
-      "types": [
-        {
-          "datatype": "string"
-        }
-      ],
-      "path": "AccessPolicy.resource.writeCriteria.pre",
-      "min": 0,
-      "max": "1",
-      "short": "",
-      "definition": "Invariant to check against the state of the resource before modification.",
-      "comment": "",
-      "inherited": false
-    },
-    {
-      "name": "post",
-      "depth": 3,
       "types": [
         {
           "datatype": "string"
         }
       ],
-      "path": "AccessPolicy.resource.writeCriteria.post",
+      "path": "AccessPolicy.resource.writeCriteria",
       "min": 0,
-      "max": "1",
+      "max": "*",
       "short": "",
-      "definition": "Invariant to check against the state of the resource with modifications tentatively applied.",
+      "definition": "Invariant that must be satisfied for the resource to be written.  Can include %before and %after placeholders to refer to the resource before and after the updates are applied.",
```

Comment:
What was your thinking behind making this an array field? Couldn't we get the same functionality by a single long expression with ORs and ANDs

Reply:
We could, but when I tried that I thought even with two conditions it got unwieldy.  Having the field be an array allows for more granular management of the individual conditions, and frees users from needing to worry about parentheses and operator precedence order most of the time

Reply:
I see. It seems like you took inspiration from the profile invariants? 

My concern is about implicit behavior. Once you have an array of conditions, you need to know if they're OR'd or AND'd together, so either have to reference the docs, or have another field specify the behavior. That's what `enableWhen` does in `Questionnaire`. But maybe if they're all _invariants_, it's clear from context that they must all be satisfied?

Reply:
I think that it should ideally be clear form context that all the criteria must be satisfied: would it help if we renamed the field to e.g. `writeConstraint` or `writeInvariant`?

Reply:
Yeah I think that's a good suggestion. `writeConstraint` maps nicely to the  overall `constraint` defined for [ElementDefinition](https://build.fhir.org/elementdefinition-definitions.html#ElementDefinition.constraint)